test-context/src/subset.rs

```diff
@@ -0,0 +1,90 @@
+use serde_json::Value;
+
+pub trait ContainsSubset {
+    // Returns true if the value is a subset of the receiver.
+    fn contains_subset(&self, value: Value) -> bool;
+    // Returns true if the value a deep subset of the receiver.
+    fn contains_deep_subset(&self, value: Value) -> bool;
+}
+
+impl ContainsSubset for Value {
+    fn contains_subset(&self, value: Value) -> bool {
+        match (self, &value) {
+            (Value::Object(src), Value::Object(subset)) => subset
+                .iter()
+                .all(|(k, v)| src.get(k).is_some_and(|x| x == v)),
+
+            (Value::Array(src), Value::Array(subset)) => {
+                subset.iter().all(|v| src.iter().any(|x| x == v))
+            }
```

Comment:
I don't think this is necessary, because `contains_deep_subset` already does this. We want to be able to pass in an array and have it match explicitly, e.g. ` m.contains_subset(json!({"purl": [ "pkg:rpm/redhat/rubygem-google-cloud-env@1.6.0-1.el8sat?arch=src" ]}))`

Reply:
This is so that: 

```rust
let actual = json!([{"a":1}]);
actual.contains_subset(json!([{"a":1, "b":2}])); // should return false
```

if we used contains_deep_subset then it would return true.

Reply:
I would expect the absence of `b` to cause that to return false. I'm not sure how that `Value::Array` branch would even come into play here. What am I missing?

Reply:
confirmed:
```rust
        let actual = json!([{"a":1}]);
        assert!(!actual.contains_subset(json!([{"a":1, "b":2}])));
        assert!(!actual.contains_deep_subset(json!([{"a":1, "b":2}])));
```
Maybe add that to `test_array_subset`?

Reply:
This works as I'd expect, too:
```rust
        let actual = json!([{"a":1, "b": 2}]);
        assert!(!actual.contains_subset(json!([{"a":1}])));
        assert!(actual.contains_deep_subset(json!([{"a":1}])));
```
Maybe `contains_partial_subset` seems best in light of those?